### PR TITLE
[feature] Reduce verbosity of interface listing and host statuses in global query output

### LINKED ISSUE #301: Reduce verbosity of interface listing and host statuses in global-query output
There are two contributors to make the global-query output undreadable:

* the collated and long list of interfaces 
* the host statuses

These can be summarized with `n interfaces` and the `m hosts queried / k errors / l ok`

### PR DESCRIPTION
**Pull request by els0r**
### Synopsis

Separates console printing into the following stages (where `[ ]` denotes optional):

* [ error logging for failed hosts in distributed query ]
* query output (the flow table)
* summary

The summary in its longest form (e.g. a distributed query with a reverse DNS lookup and a tracing infrastructure in place) will look like:

```
Timespan           : [2024-04-15 20:44:35, 2024-04-15 21:46:37] (1h2m0s)
Interfaces / Hosts : 280 queried on 248 hosts: 114 ok / 0 empty / 134 error
Sorted by          : accumulated data volume (sent and received)
Query stats        : displayed top 25 hits out of 827 in 13.32s
Conditions         : dport = 443
Trace ID           : cff4115b6080bce9c0c233b560acd71e
```

### Changes

* Interface names are only printed if the query attribute doesn't include the interface
* removes the reverse DNS stats. The information can be extracted from a span in case a tracing infrastructure is available

### Additions

* the host statuses are summarized under the `Hosts` key
* the `Trace ID` is included if it is available to allow callers of a distributed query to inspect / debug the call across the infrastructure

Closes #301

### CODE REVIEW

File: pkg/query/dns/dns.go [hunk 1]
```diff
@@ -68,7 +77,7 @@ func TimedReverseLookup(ips []string, timeout time.Duration) (ipToDomain map[str
 			if LookupResult.Success {
 				ipToDomain[LookupResult.IP] = LookupResult.Domain
 			}
-		case <-time.After(timeout):
+		case <-queryCtx.Done(): // timeout case
 			pending = 0
```
Comment (fako1024) [inline pkg/query/dns/dns.go:81]: Nitpick: Why not just `return` here? using the `pending` variable as abort criterion feels a bit non-idiomatic...

File: pkg/results/TablePrinter.go [hunk 2]
```diff
@@ -637,19 +650,31 @@ func (t *TextTablePrinter) Footer(result *Result) error {
...
 			formatting.Durationable(result.Summary.Timings.ResolutionDuration),
-			formatting.Durationable(t.resolveTimeout))
+			formatting.Durationable(t.resolveTimeout),
+		)
 	}
 
+	// only print interface names if the attributes don't include the interface
+	var (
+		ifaceSummary string
+		ifaceKey     = "Interface"
```
Comment (fako1024) [inline pkg/results/TablePrinter.go:664]: Could use a constant I guess?

File: pkg/results/TablePrinter.go [hunk 3]
```diff
@@ -637,19 +650,31 @@ func (t *TextTablePrinter) Footer(result *Result) error {
...
+	)
+	if len(result.Summary.Interfaces) > 1 {
+		ifaceKey += "s"
+	}
+	if t.basePrinter.selector.Iface {
+		ifaceSummary = result.Summary.Interfaces.Summary() + " queried"
+	} else {
+		ifaceSummary = strings.Join(result.Summary.Interfaces, ",")
+	}
+	t.footerWriter.WriteEntry(ifaceKey, ifaceSummary)
```
Comment (fako1024) [inline pkg/results/TablePrinter.go:674]: As discussed: I'd merge interfaces / hosts on one line (if both exist)

File: pkg/results/TablePrinter.go [hunk 4]
```diff
@@ -637,19 +650,31 @@ func (t *TextTablePrinter) Footer(result *Result) error {
...
-		result.Summary.First.Format(types.DefaultTimeOutputFormat),
-		result.Summary.Last.Format(types.DefaultTimeOutputFormat),
-		formatting.Durationable(result.Summary.Last.Sub(result.Summary.First).Round(time.Minute)),
-		strings.Join(result.Summary.Interfaces, ","))
-	fmt.Fprintf(t.footwriter, "Sorted by\t: %s\n",
-		describe(t.sort, t.direction))
+	result.Summary.TimeRange.PrintFooter(t.footerWriter)
 	if result.Summary.Timings.ResolutionDuration > 0 {
-		fmt.Fprintf(t.footwriter, "Reverse DNS stats\t: RDNS took %s, timeout was %s\n",
+		t.footerWriter.WriteEntry("Reverse DNS stats", "RDNS took %s, timeout was %s",
```
Comment (fako1024) [inline pkg/results/TablePrinter.go:655]: As discussed: Let's get rid of this and rely on instrumentation...

File: pkg/results/print.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+// FooterPrinter allows a type to print to the Footer
+type FooterPrinter interface {
+	PrintFooter(fw *FooterTabwriter) error
+}
+
+/// FooterPrinter implementations
+
+func (tr TimeRange) PrintFooter(fw *FooterTabwriter) error {
```
Comment (fako1024) [inline pkg/results/print.go:43]: Nitpick: These should complain about missing documentation on the exposed methods.

File: .gitignore [hunk 6]
```diff
@@ -11,4 +11,4 @@ cmd/legacy/legacy
 .vscode/
 _build/
 go.work.sum
-
+.db/
```
[the remark this answers is not in the document]
Comment (els0r) [reply .gitignore:14]: That's correct. For a good reason: I'm developing using the `devcontainer` and whatever is inside the directory will also be mounted into the running docker container.

File: pkg/query/dns/dns.go (hunk 1 above)
Comment (els0r) [reply pkg/query/dns/dns.go:81]: Not sure. Not my code :P. Allow me to refactor once I move ahead with the augmentation plugins upon printing.

File: pkg/results/TablePrinter.go [hunk 7]
```diff
@@ -637,19 +650,31 @@ func (t *TextTablePrinter) Footer(result *Result) error {
...
+		)
 	}
 
+	// only print interface names if the attributes don't include the interface
+	var (
+		ifaceSummary string
+		ifaceKey     = "Interface"
+	)
+	if len(result.Summary.Interfaces) > 1 {
+		ifaceKey += "s"
```
[the remark this answers is not in the document]
Comment (els0r) [reply pkg/results/TablePrinter.go:667]: Call me pedantic, but it always bugs me to see `Interfaces` and then a 1.

File: pkg/results/TablePrinter.go (hunk 2 above)
Comment (els0r) [reply pkg/results/TablePrinter.go:664]: true

File: pkg/results/result.go [hunk 8]
```diff
@@ -192,16 +217,39 @@ func (hs HostsStatuses) Print(w io.Writer) error {
 		case types.StatusError:
 			withError++
 		}
-		hosts = append(hosts, struct {
-			host string
-			Status
-		}{host: host, Status: status})
+	}
+	return fmt.Sprintf("%d total: %d ok / %d empty / %d error", len(hs), ok, empty, withError)
```
Comment (els0r) [inline pkg/results/result.go:221]:
```suggestion
	return fmt.Sprintf("%d hosts: %d ok / %d empty / %d error", len(hs), ok, empty, withError)
```

**Pull request merged**